exaudfclient/base/javacontainer/script_options/parser_ctpg_script_importer.cc

```diff
@@ -0,0 +1,87 @@
+#include "base/javacontainer/script_options/parser_ctpg_script_importer.h"
+#include "base/swig_factory/swig_factory.h"
+#include "base/utils/debug_message.h"
+#include "base/utils/exceptions.h"
+#include "base/script_options_parser/exception.h"
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+
+namespace ctpg_parser = ExecutionGraph::OptionsLineParser::CTPG;
+
+namespace SWIGVMContainers {
+
+namespace JavaScriptOptions {
+
+namespace CTPG {
+
+ScriptImporter::ScriptImporter(SwigFactory & swigFactory, Keywords & keywords)
+: m_importedScriptChecksums()
+, m_swigFactory(swigFactory)
+, m_metaData()
+, m_keywords(keywords) {}
+
+void ScriptImporter::importScript(std::string & scriptCode, ctpg_parser::options_map_t & options) {
+    const auto optionIt = options.find(std::string(m_keywords.importKeyword()));
+    if (optionIt != options.end()) {
+        m_importedScriptChecksums.addScript(scriptCode.c_str());
+        //Sort options from first in script to last in script
+        std::sort(optionIt->second.begin(), optionIt->second.end(),
```

Comment:
This is inplace sorting or

Reply:
Thats why also a non const ref for options

Reply:
Exactly